src/data.js

```diff
@@ -348,6 +348,63 @@ let Artifact = Entity.configure({
     'publicBucket',   // Public artifact bucket wrapping S3
     'monitor',        // base.monitor instance
   ],
+}).configure({
+  version: 2,
+  properties: {
+    taskId:         Entity.types.SlugId,
+    runId:          Entity.types.Number,
+    name:           Entity.types.String,
+    storageType:    Entity.types.String,
+    contentType:    Entity.types.String,
+    /**
+     * Location details storageType,
+     *
+     * storageType: s3
+     *   bucket:        S3 bucket that contains the object
+     *   prefix:        Prefix (path) for the object within the bucket
+     *
+     * storageType: azure
+     *   container:     Azure container that holds the blob
+     *   path:          Path to blob within container
+     *
+     * storageType: reference
+     *   url:           URL that artifact should redirect to
+     *
+     * storageType: error
+     *   reason:        Formalized reason for error artifact, see JSON schema.
+     *   message:       Human readable error message to return
+     * storageType: blob
+     *   contentType
+     *   contentEncoding
+     *   contentSha256: SHA256 hash of the un-encoded artifact
+     *   contentLength: Number of bytes of the un-encoded artifact
+     *   transferSha256: SHA256 hash of the content-encoding encoded artifact
+     *   transferLength: Number of bytes of the content-encoding encoded artifact
+     *   partsHash: Rather than storing the potentially large list of sha256/size
+     *              pairings for each part, we just need to store enough information
+     *              to determine if the operation would result in an idempotency
+     *              error
+     */
```

Comment:
Please document the `bucket` and `key` properties... anything else we're missing? Please check it's super hard to maintain this if we forget what properties we're storing.